test/user_test.rb

```diff
@@ -0,0 +1,46 @@
+require 'minitest/autorun'
+require 'minitest/pride'
+require './lib/joke'
+require './lib/user'
+
+class UserTest < Minitest::Test
+  def test_user_exists
+    sal = User.new("Sal")
+    assert_instance_of User, sal
+  end
+
+  def test_can_learn_joke
+    sal = User.new("Sal")
+    joke_1 = Joke.new(1, "Why did the strawberry cross the road?", "Because his mother was in a jam.")
+    joke_2 = Joke.new(2, "How do you keep a lion from charging?", "Take away its credit cards.")
+    sal.learn(joke_1)
+    sal.learn(joke_2)
+    assert_equal sal.jokes.length, 2
+  end
+
+  def test_can_tell_joke
+    sal = User.new("Sal")
+    ali = User.new("Ali")
+    joke_1 = Joke.new(1, "Why did the strawberry cross the road?", "Because his mother was in a jam.")
+    joke_2 = Joke.new(2, "How do you keep a lion from charging?", "Take away its credit cards.")
+    sal.learn(joke_1)
+    sal.learn(joke_2)
+    sal.tell(ali, joke_1)
+    sal.tell(ali, joke_2)
+
+    assert_equal sal.jokes, ali.jokes
```

Comment:
this is clever! i like the idea behind it.. although what if they were both empty arrays? technically, this assertion would still pass in that case.  You'd need to also assert that sal had an array of the right jokes _before_ making a second assertion that sal.jokes equals ali.jokes to make sure that they weren't just empty arrays, or something else incorrect. 

It's probably easiest to just `assert_equal [joke_1, joke_2], ali.jokes` (technically, you've already tested the `learn` method, so you don't need to check his jokes in the test if you're checking that ali has the right jokes)